gateway/src/apicast/proxy.lua

```diff
@@ -372,6 +380,17 @@ function _M:handle_backend_response(cached_key, response, ttl)
   local authorized = (response.status == 200)
   local unauthorized_reason = not authorized and rejection_reason(response.headers)
   local retry_after = not authorized and limit_reset(response.headers)
+  local limit_max = limit_max_value(response.headers)
+
+  -- This is for disabled metrics. Those have a limit of 0 in the 3scale
+  -- backend. When authorizing a disabled metric, backend returns "limits
+  -- exceeded" as the unauthorized reason. However, from the point of view of
+  -- APIcast, we want to distinguish between limits exceeded vs disabled
+  -- metric. That's why we reset the reason. The generic auth fail (403) will
+  -- be returned.
+  if tonumber(limit_max) == 0 and unauthorized_reason == 'limits_exceeded' then
```

Comment:
If we return the number in the limit_max_value, here tonumber is not needed at all.

Reply:
Fixed :+1: